tests/std/tests/P0896R4_ranges_alg_unique_copy/test.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <span>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+using namespace std;
+using P = pair<int, int>;
+
+// Validate that unique_copy_result aliases in_out_result
+STATIC_ASSERT(same_as<ranges::unique_copy_result<int, double>, ranges::in_out_result<int, double>>);
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::unique_copy(borrowed<false>{}, nullptr_to<int>)),
+    ranges::unique_copy_result<ranges::dangling, int*>>);
+STATIC_ASSERT(
+    same_as<decltype(ranges::unique_copy(borrowed<true>{}, nullptr_to<int>)), ranges::unique_copy_result<int*, int*>>);
+
+struct instantiator {
+    static constexpr P expected[4] = {{0, 99}, {1, 47}, {3, 99}, {4, 47}};
+    static constexpr P input[6]    = {{0, 99}, {1, 47}, {2, 47}, {3, 99}, {4, 47}, {5, 47}};
+
+    template <ranges::input_range Read, weakly_incrementable Write>
+    static constexpr void call() {
+        using ranges::unique_copy, ranges::unique_copy_result, ranges::equal, ranges::equal_to, ranges::iterator_t;
+
+        { // Validate iterator + sentinel overload
+            P output[4] = {{-1, -1}, {-1, -1}, {-1, -1}, {-1, -1}};
+            Read wrapped_input{input};
+
+            // Tests which implementation strategy was choosen
+            size_t inputCounter    = 0;
+            size_t outputCounter   = 0;
+            size_t storeCounter    = 0;
+            auto countedProjection = [&](const P& val) {
+                if (output <= addressof(val) && addressof(val) < output + 4) {
+                    ++outputCounter;
+                } else if (input <= addressof(val) && addressof(val) < input + 6) {

```

Comment:
I believe that `end()` would be clearer, so we don't have to verify that these are indeed the sizes:
```suggestion
                if (output <= addressof(val) && addressof(val) < end(output)) {
                    ++outputCounter;
                } else if (input <= addressof(val) && addressof(val) < end(input)) {
```